addons/mass_mailing/controllers/portal.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from werkzeug.exceptions import NotFound
+
+from odoo.http import Controller, request, route
+from odoo.tools import consteq
+
+
+class MassMailingController(Controller):
+
+    @route('/mass_mailing/turnoff_report', type='http', website=True, auth='user')
+    def turn_off_mailing_reports(self, token=None):
+        if consteq(token, request.env['mass_mailing.mailing']._get_unsubscribe_token()):
+            try:
+                request.env['ir.config_parameter'].set_param('mass_mailing.mass_mailing_reports', False)
```

Comment:
will this work if I'm a mailing user but not the admin ? I think you should use sudo and explicitly check that the user is a mailing user